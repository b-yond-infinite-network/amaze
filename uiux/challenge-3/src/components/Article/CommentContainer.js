import CommentInput from './CommentInput';
import CommentList from './CommentList';
import { Link } from 'react-router-dom';
import React from 'react';

const CommentContainer = props => {
  if (props.currentUser) {
    return (
      <div id="comment-container" className="col-xs-12 col-md-8 offset-md-2">
        <div>
          <list-errors errors={props.errors}></list-errors>
          <CommentInput id="comment-input" slug={props.slug} currentUser={props.currentUser} />
        </div>

        <CommentList
          id="comment-list"
          comments={props.comments}
          slug={props.slug}
          currentUser={props.currentUser} />
      </div>
    );
  } else {
    return (
      <div id="comment-container" className="col-xs-12 col-md-8 offset-md-2">
        <p>
          <Link id="comment-login" to="/login">Sign in</Link>
          &nbsp;or&nbsp;
          <Link id="comment-register" to="/register">sign up</Link>
          &nbsp;to add comments on this article.
        </p>

        <CommentList
          id="comment-list"
          comments={props.comments}
          slug={props.slug}
          currentUser={props.currentUser} />
      </div>
    );
  }
};

export default CommentContainer;
