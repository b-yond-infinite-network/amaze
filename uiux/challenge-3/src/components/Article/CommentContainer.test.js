import CommentContainer from './CommentContainer';
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../../store'

const initialState = {
    currentUser: {
        username: "test username"
    },
    errors: "",
    slug: "test slug",
    comments: [
        {
            id: '1234',
            body: 'Test body',
            author: {
                username: 'test username',
                image: ''
            },
            createdAt: '2019-06-29T18:36:23.216Z'
        },
        {
            id: '12345',
            body: 'Test body 2',
            author: {
                username: 'test username 2',
                image: ''
            },
            createdAt: '2019-06-30T18:36:23.216Z'
        }
    ]
};

describe('CommentContainer Component', () => {

    let mockStore;
    let comment;
    let currentUser;

    let tree;
    let wrapper;
    let render;

    describe('has currentUser', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)({ initialState });
            comment = initialState.comment;
            
            const props = initialState;

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <CommentContainer {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = shallow(tree).childAt(0).childAt(0).dive();
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should render without errors', () => {
            const component = wrapper.find('#comment-container');
            expect(component.length).toBe(1);
        });

        it('Should render comment input', () => {
            const component = wrapper.find('#comment-input');
            expect(component.length).toBe(1);
        });

        it('Should render comment list', () => {
            const component = wrapper.find('#comment-list');
            expect(component.length).toBe(1);
        });

    });

    describe('does NOT have currentUser', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)({ initialState });
            comment = initialState.comment;
            
            const props = initialState;
            props.currentUser = null;

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <CommentContainer {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = shallow(tree).childAt(0).childAt(0).dive();
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should render without errors', () => {
            const component = wrapper.find('#comment-container');
            expect(component.length).toBe(1);
        });

        it('Should NOT render comment input', () => {
            const component = wrapper.find('#comment-input');
            expect(component.length).toBe(0);
        });

        it('Should render login', () => {
            const component = wrapper.find('#comment-login');
            expect(component.length).toBe(1);
        });

        it('Should render register', () => {
            const component = wrapper.find('#comment-register');
            expect(component.length).toBe(1);
        });

        it('Should render comment list', () => {
            const component = wrapper.find('#comment-list');
            expect(component.length).toBe(1);
        });

    });

});