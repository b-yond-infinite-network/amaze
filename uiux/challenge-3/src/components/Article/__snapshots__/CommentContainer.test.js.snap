// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentContainer Component does NOT have currentUser Should match the snapshot 1`] = `
<div
  className="col-xs-12 col-md-8 offset-md-2"
  id="comment-container"
>
  <p>
    <a
      href="/login"
      id="comment-login"
      onClick={[Function]}
    >
      Sign in
    </a>
     or 
    <a
      href="/register"
      id="comment-register"
      onClick={[Function]}
    >
      sign up
    </a>
     to add comments on this article.
  </p>
  <div>
    <div
      className="card"
      id="comment-card"
    >
      <div
        className="card-block"
      >
        <p
          className="card-text"
        >
          Test body
        </p>
      </div>
      <div
        className="card-footer"
      >
        <a
          className="comment-author"
          href="/@test username"
          id="comment-author-image"
          onClick={[Function]}
        >
          <img
            alt="test username"
            className="comment-author-img"
            src=""
          />
        </a>
         
        <a
          className="comment-author"
          href="/@test username"
          id="comment-author"
          onClick={[Function]}
        >
          test username
        </a>
        <span
          className="date-posted"
          id="comment-created"
        >
          Sat Jun 29 2019
        </span>
      </div>
    </div>
    <div
      className="card"
      id="comment-card"
    >
      <div
        className="card-block"
      >
        <p
          className="card-text"
        >
          Test body 2
        </p>
      </div>
      <div
        className="card-footer"
      >
        <a
          className="comment-author"
          href="/@test username 2"
          id="comment-author-image"
          onClick={[Function]}
        >
          <img
            alt="test username 2"
            className="comment-author-img"
            src=""
          />
        </a>
         
        <a
          className="comment-author"
          href="/@test username 2"
          id="comment-author"
          onClick={[Function]}
        >
          test username 2
        </a>
        <span
          className="date-posted"
          id="comment-created"
        >
          Sun Jun 30 2019
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentContainer Component has currentUser Should match the snapshot 1`] = `
<div
  className="col-xs-12 col-md-8 offset-md-2"
  id="comment-container"
>
  <div>
    <list-errors
      errors=""
    />
    <form
      className="card comment-form"
      onSubmit={[Function]}
    >
      <div
        className="card-block"
      >
        <textarea
          className="form-control"
          onChange={[Function]}
          placeholder="Write a comment..."
          rows="3"
          value=""
        />
      </div>
      <div
        className="card-footer"
      >
        <img
          alt="test username"
          className="comment-author-img"
        />
        <button
          className="btn btn-sm btn-primary"
          type="submit"
        >
          Post Comment
        </button>
      </div>
    </form>
  </div>
  <div>
    <div
      className="card"
      id="comment-card"
    >
      <div
        className="card-block"
      >
        <p
          className="card-text"
        >
          Test body
        </p>
      </div>
      <div
        className="card-footer"
      >
        <a
          className="comment-author"
          href="/@test username"
          id="comment-author-image"
          onClick={[Function]}
        >
          <img
            alt="test username"
            className="comment-author-img"
            src=""
          />
        </a>
         
        <a
          className="comment-author"
          href="/@test username"
          id="comment-author"
          onClick={[Function]}
        >
          test username
        </a>
        <span
          className="date-posted"
          id="comment-created"
        >
          Sat Jun 29 2019
        </span>
        <span
          className="mod-options"
          id="delete-button"
        >
          <i
            className="ion-trash-a"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
    <div
      className="card"
      id="comment-card"
    >
      <div
        className="card-block"
      >
        <p
          className="card-text"
        >
          Test body 2
        </p>
      </div>
      <div
        className="card-footer"
      >
        <a
          className="comment-author"
          href="/@test username 2"
          id="comment-author-image"
          onClick={[Function]}
        >
          <img
            alt="test username 2"
            className="comment-author-img"
            src=""
          />
        </a>
         
        <a
          className="comment-author"
          href="/@test username 2"
          id="comment-author"
          onClick={[Function]}
        >
          test username 2
        </a>
        <span
          className="date-posted"
          id="comment-created"
        >
          Sun Jun 30 2019
        </span>
      </div>
    </div>
  </div>
</div>
`;
